### DISCUSSION TITLE
[Question] Observe Method to CanExecute ReactiveCommand

### DISCUSSION THREAD

**Discussion by Tenjim** [reactiveui/ReactiveUI#2804] [Q&A] [answered]
Hi all,

I'm working in WPF App and Avalonia.
I wanna to ask you if there is a possibility to observe a method to canExecute ReactiveCommand ?
I had before a implementation with a RelayCommand

```
public class TestViewModel 
{
      private string _messageTest = "Hello";
      public string MessageTest
      {
          get => _messageTest;
          set => this.RaiseAndSetIfChanged(ref _messageTest, value);
      }
      
      private bool _canExecucte;
      public bool CanExecute
      {
          get => _canExecucte;
          set => this.RaiseAndSetIfChanged(ref _canExecucte, value);
      }
      public ReactiveCommand<Unit, Unit> TestCommand { get; set; }
      
      public TestViewModel()
      {
           TestCommand = ReactiveCommand.Create(Hello, ()=> isValidMessage());
      }

      public void isValidMessage()
      {
           return MessageTest.length>10;
      }
      public void Hello()
      {
           MessageTest="Test";
      }

}
```
In this sample : I want to bind IsValidMessage. I know that ReactiveUi use WhenAnyValue when we want to observe properties changed.
Just in my case I want to observe method.

Thank in advance,

**Comment by glennawatson** [comment, marked as answer]
You can do 

```cs
var canExecute = this.WhenAnyValue(x => x.MessageText).Select(x => x?.Length > 10)
```

You could also use a method like

```cs
var canExecute = this.WhenAnyValue(x => x.MessageText).Select(x => IsValidMessage(x));

private bool IsValidMessage(string message) 
{
   return message?.Length > 10;
}
```


Also use the GitHub discussions tab in the future if you have questions rather than a issue.

**Reply by Tenjim** [reply]
Thank to your fast reply and sorry I'll use now the discussion tab when I have question like this.

Yes I doing as a workaround something like you in WhenAnyValue.
Is it mandatory to observe properties in WhenAnyValue ?

**Reply by heronbpv** [reply]
Not really. It's just a convenience.
The canExecute parameter of the Create method is itself a IObservable of bool, so any observable that returns a boolean qualifies.

**Reply by Tenjim** [reply]
Ok, I'll doing so something like your answers !

Thank for your answer
